vm/src/value.rs

```diff
@@ -270,7 +472,7 @@ impl Value {
         match &self {
             Value::Byte(b) => Ok(*b as f32),
             Value::Int(i) => Ok(from_i56(i) as f32),
-            Value::Float(f) => Ok(f.0),
+            Value::Float(f) => Ok(f32::from(*f)),
```

Comment:
I used f32 here because `cargo test` was not passing the `test_pol` function with f64. I'm realizing now what the issue is and it has wider implications. Here's the test result:
```
---- vm::tests::test_pol stdout ----
thread 'vm::tests::test_pol' panicked at vm/src/vm.rs:914:9:
assertion `left == right` failed
  left: 12499.999999998836
 right: 12500.0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
The F56 only intends to have 12-14 digits of decimal precision and this 12499.999999998836 result actually has about that many good digits, but because we convert the F56 to an f64, the function that converts an f64 to a string expects a few extra digits of precision that the F56 just didn't have. So pretty much anywhere convert F56 to f64, the resulting f64 won't properly convert to a decimal string. The same thing happens if we convert an f32 to an f64
```
        let v: f32 = 4.8;
        println!("{:?}", v); // 4.8
        println!("{:?}", v as f64); // 4.800000190734863
```

The proper solution depends on the context of when / how often we convert from Value back to primitives. We could always round before we convert to f64, but I'd like to avoid that if possible. If the string representation only matters for our tests then we can just convert to f32 to check the string. Let me know what you think.

Somewhat related, I already implemented a custom Display trait for F56 that only relies on the 12 decimal digits of precision because of the `test_tok_floats` test in `reader.rs`.

Reply:
Bump. Any thoughts on this?

Reply:
It would be good to not have the weird precision stuff show up like this.  If I understand this correctly then if we use the Display on F56 directly (ie don't convert to a f64 first) then it will work fine?  If so I think we should make sure that the lisp code that displays floats should do this.  For tests it is probably fine to convert to f32 as long as that won't loose anything we need otherwise we need to probably use Eq on F56 that I think takes care of this with epsilon now?

Reply:
Absolutely correct, F56's Display should work fine. And for displaying the lisp values, the 'display_value' function in value.rs looks to appropriately use the F56 Display.